lib/book_settings.py

```diff
@@ -144,12 +143,7 @@ def search_for_asin(self, query):
         except Exception as e:
             try:
                 self._aConnection.close()
-                if self._proxy:
-                    self._aConnection = HTTPSConnection(self._https_address, self._https_port)
-                    self._aConnection.set_tunnel('www.amazon.com', 443)
-                else:
-                    self._aConnection = HTTPSConnection('www.amazon.com')
-
+                self._aConnection.connect()
```

Comment:
I'm now confused as to when we'll use the close() at all.. In `HttpConnection.send()` if a send fails due to `BrokenPipe` then `.close()` is called - and as `auto_open` is always set we'll immediately try again.

Do you think we've hit connection problems in the past that (implicitly) the auto-close/reopen hasn't fixed, but that forceably doing it ourself has helped?